src/main/java/org/terasology/additionalitempipes/action/UnificatorAction.java

```diff
@@ -0,0 +1,144 @@
+import org.terasology.registry.In;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.BlockManager;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class UnificatorAction extends BaseComponentSystem {
+    private Side outputside;
+    @In
+    private BlockManager blockManager;
+    @In
+    PipeSystem pipeSystem;
+    @In
+    private WorldProvider world;
+    @In
+    UnificatorComponent unificatorComponent;
+
+    @ReceiveEvent(components = {UnificatorComponent.class})
+    public void detectSide(ActivateEvent event, EntityRef itemEntity) {
+        if (!event.getTarget().exists()) {
+            event.consume();
+            return;
+        }
+        Vector3i posi = event.getTarget().getComponent(BlockComponent.class).getPosition();
+        Side side = Side.inDirection(event.getHitNormal());
+
+        switch (side) {
+            case TOP:
+                outputside = Side.TOP;
+                Block top = blockManager.getBlock("AdditionalItemPipes:unificatortop");
+                world.setBlock(posi, top);
+                break;
+            case FRONT:
+                outputside = Side.FRONT;
+                Block front = blockManager.getBlock("AdditionalItemPipes:unificatorfront");
+                world.setBlock(posi, front);
+                break;
+            case RIGHT:
+                outputside = Side.RIGHT;
+                Block right = blockManager.getBlock("AdditionalItemPipes:unificatorright");
+                world.setBlock(posi, right);
+                break;
+            case BACK:
+                outputside = Side.BACK;
+                Block back = blockManager.getBlock("AdditionalItemPipes:unificatorback");
+                world.setBlock(posi, back);
+                break;
+            case LEFT:
+                outputside = Side.LEFT;
+                Block left = blockManager.getBlock("AdditionalItemPipes:unificatorleft");
+                world.setBlock(posi, left);
+                break;
+            case BOTTOM:
+                outputside = Side.BOTTOM;
+                Block bottom = blockManager.getBlock("AdditionalItemPipes:unificatorbottom");
+                world.setBlock(posi, bottom);
+                break;
+        }
+    }
+
+    @ReceiveEvent(components = UnificatorComponent.class, priority = EventPriority.PRIORITY_HIGH)
+    public void onItemInput(PipeInsertEvent event, EntityRef entity, UnificatorComponent unificator, BlockComponent block) {
+        EntityRef item = event.getActor();
+        item.removeComponent(RigidBodyComponent.class);
+        item.removeComponent(LifespanComponent.class);
+        item.removeComponent(PickupComponent.class);
+        Vector3i position = block.getPosition();
+        if(outputside != null) {
```

Comment:
`Side.values()` returns all possible sides (in an array) in order like they're defined in `Side.class`. Instead of using the `num` and getting the `Side` from `Side.values()` array, just pass the `outputside` as `inputOrDrop`'s parameter.